clang/lib/DPCT/DPCT.cpp

```diff
@@ -498,6 +498,54 @@ static void loadMainSrcFileInfo(clang::tooling::UnifiedPath OutRoot) {
   }
 }
 
+std::vector<std::string> SplitStr(const std::string &Str, char Delimiter) {
+  std::vector<std::string> Result;
+  std::istringstream Stream(Str);
+  std::string Token;
+  while (std::getline(Stream, Token, Delimiter)) {
+    Result.push_back(Token);
+  }
+  return Result;
+}
+
+std::string
+GetFirstAvailableTool(std::unordered_set<std::string> &ToolCandidates) {
+  const char *EnvPath = std::getenv("PATH");
```

Comment:
Try using `std::string::find` to extract `path` and then check available one by one.